### DISCUSSION TITLE
How to represent object in literal?

### DISCUSSION THREAD

**Discussion by Xuanwo** [databendlabs/databend#8655] [Q&A] [answered]
I can represent array/tuple by

```
SELECT ('a', 1, true); -> tuple
SELECT [1, 2, 3]; -> array
```

But how to present an object in literal without `parse_json`? With this support, I can implement something like:

```sql
my_table_function({
  'arg1': 123
})
```

cc @andylokandy & @b41sh  for help.

**Comment by Xuanwo** [comment]
Maybe we can implement an `Object` in `Expr`?

**Comment by andylokandy** [comment]
It's not supported yet but we can make it a syntax sugar if desired.

**Comment by b41sh** [comment, marked as answer]
new expression add a new data type `map`, maybe we can convert `{'arg1': 123}` to map.

**Reply by Xuanwo** [reply]
`{'arg1': 123}` => map, and `{arg1 => 123}` to object?

**Reply by b41sh** [reply]
We will remove `Array` and `Object` data types in the future, and use `Variant` instead.
And we can support converting string literals like `"{\"arg1\": 123}"` to `Variant` without the `parse_json` function.
